docs/guides/latest/volumesnapshot/deployment.md

```diff
@@ -0,0 +1,493 @@
+```
+You will also see that, The YAML for `backup-pvc-1-1560662826` looks like using the following command,
+
+```console
+$ kubectl get volumesnapshot backup-pvc-1-1560662826 -n demo -o yaml
+apiVersion: snapshot.storage.k8s.io/v1alpha1
+kind: VolumeSnapshot
+metadata:
+  creationTimestamp: "2019-06-16T05:27:07Z"
+  finalizers:
+  - snapshot.storage.kubernetes.io/volumesnapshot-protection
+  generation: 4
+  name: backup-pvc-1-1560662826
+  namespace: demo
+  resourceVersion: "751269"
+  selfLink: /apis/snapshot.storage.k8s.io/v1alpha1/namespaces/demo/volumesnapshots/backup-pvc-1-1560662826
+  uid: 6206ec6b-8ff7-11e9-bd3e-42010a800011
+spec:
+  snapshotClassName: default-snapshot-class
+  snapshotContentName: snapcontent-6206ec6b-8ff7-11e9-bd3e-42010a800011
+  source:
+    apiGroup: null
+    kind: PersistentVolumeClaim
+    name: backup-pvc-1
+status:
+  creationTime: "2019-06-16T05:27:08Z"
+  readyToUse: true
+  restoreSize: 6Gi
+```
+
+The snapshots will be stored in the GCE cloud.
+You                 can view the snapshot in GCE,
+<p align="center">
+  <img alt="Stash Backup Flow" src="/docs/images/v1beta1/backends/volumesnapshot/deployment.png">
+<figcaption align="center">Fig: Snapshots Screenshot</figcaption>
+</p>
+
+Once the snapshotting is completed, `BackupSession` crd phase will be Succeeded. Check the `BackupSession` phase is Succeeded by following command,
+
+```console
+$ kubectl get bs -n demo -l  backup-configuration=deployments-volume-snapshot
+NAME                                     BACKUPCONFIGURATION           PHASE       AGE
+deployments-volume-snapshot-1559026686   deployments-volume-snapshot   Succeeded   40s
+```
+
+### Restore PVC from VolumeSnapshot
+
+This section will show you how to restore the PVCs from the snapshots we have taken in earlier section.
+
+#### Create RestoreSession
+
+Create a `RestoreSession` crd to restore PersistentVolumeClaims from snapshot.
+Sample `RestoreSession` crd YAML are given below,
+
+```yaml
+apiVersion: stash.appscode.com/v1beta1
+kind: RestoreSession
+metadata:
+  name: restore-pvc
+  namespace: demo
+spec:
+  driver: VolumeSnapshotter
+  target:
+    volumeClaimTemplates:
+      - metadata:
+          name: backup-pvc
+        spec:
+          accessModes: [ "ReadWriteOnce" ]
+          storageClassName: "standard"
+          resources:
+            requests:
+              storage: 6Gi
+          dataSource:
+            kind: VolumeSnapshot
+            name: ${CLAIM_NAME}-1559026686
+           # name: backup-pvc-1559026686
+            apiGroup: snapshot.storage.k8s.io
+```
+Here,
+
+* `spec.driver` indicates the name of the agent to use to back up the target. Currently, Stash supports `Restic`, `VolumeSnapshotter` drivers. The `VolumeSnapshotter` is used to backup/restore PVC using `VolumeSnapshot` API.
+
+* `spec.target.volumeClaimTemplates`:
+  * `metadata.name` is the name of the restored `PVC` or prefix of the `VolumeSnapshot` name.
+  * `spec.dataSource`:
```

Comment:
`spec.dataSource`: `spec.dataSource` specifies the source of the data from where the newly created PVC will be intialized. It requires following fields to set: